src/backend/utils/adt/datetime.c

```diff
@@ -26,6 +26,7 @@
 #include "funcapi.h"
 #include "miscadmin.h"
 #include "nodes/nodeFuncs.h"
+#include "parser/parser.h"
```

Comment:
this is not the ideal approach. We cannot modify PG behaviour for any PG data types. Let us start with a document to describe the data type -> behavior differences and relevant functions which handle the logic. We have to provide  hooks at the right places to override PG behaviour with Babel behavior conditionally. Depending on the number of changes required, we can decide the granularity of hooks.